crates/bevy_render/src/render_asset.rs

```diff
@@ -26,16 +26,22 @@ pub trait RenderAsset: Asset {
     type ExtractedAsset: Send + Sync + 'static;
     /// The GPU-representation of the the asset.
     type PreparedAsset: Send + Sync + 'static;
+    /// Specifies all ECS data required by [`RenderAsset::extract_asset`].
+    /// For convenience use the [`lifetimeless`](bevy_ecs::system::lifetimeless) SystemParams.
+    type ExtractParam: SystemParam;
     /// Specifies all ECS data required by [`RenderAsset::prepare_asset`].
     /// For convenience use the [`lifetimeless`](bevy_ecs::system::lifetimeless) SystemParams.
-    type Param: SystemParam;
-    /// Converts the asset into a [`RenderAsset::ExtractedAsset`].
-    fn extract_asset(&self) -> Self::ExtractedAsset;
+    type PrepareParam: SystemParam;
+    /// Converts the asset into a [`RenderAsset::ExtractedAsset`]. Therefore ECS data may be accessed via the `param`.
+    fn extract_asset(
+        &self,
+        param: &mut SystemParamItem<Self::ExtractParam>,
+    ) -> Self::ExtractedAsset;
```

Comment:
Can I also go ahead and change this to: `Result<Self::ExtractedAsset, ExtractAssetError<Self::ExtractedAsset>>`? This is useful because we can block the render resource creation if the extract doesn't mean X criteria.

Reply:
I went ahead and implemented this because it makes sense to me!